lib/json_api/json_api.rb

```diff
@@ -0,0 +1,70 @@
+require 'sinatra/base'
+require 'sinatra/json'
+require 'sinatra/namespace'
+require 'json_api/game_deserializer'
+require 'json_api/game_serializer'
+require 'json_api/player_deserializer'
+require 'game/game'
+
+module JsonAPI
+  class JsonAPI < Sinatra::Base
+    register Sinatra::Namespace
+
+    def initialize
+      super
+      @player_deserializer = PlayerDeserializer.new
+      @game_serializer = GameSerializer.new
+      @game_deserializer = GameDeserializer.new
+    end
+
+    namespace '/api/v1' do
+      new_game = lambda do
+        begin
+          req_body = request_body
+          players = @player_deserializer.deserialize(req_body["players"])
+          game = Game::Game.new(players, board_size: board_size_or(req_body, 3))
+          json @game_serializer.serialize(game)
+        rescue
+          json "error" => "invalid players provided"
+        end
+      end
+
+      play_game = lambda do
```

Comment:
Would it make sense to pull these out  into straight functions? Don't often see lambdas like this in ruby for this purpose

Reply:
Usually I see logic like this pulled out into a separate file or just separated out so the controller method itself is slim